### ISSUE TITLE
stop writing Javascript!

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-web-ide-backend#83]
It's total nonsense that we're writing this in The Worst Programming Language Ever when we we have a whole programming language that runs on both client and server side and has a proper IDE and is way way more maintainable and expressive.

**Comment by quintesse** [comment]
High prio? Really now?

**Comment by gavinking** [comment]
Absolutely! It's a huge problem that we don't have complete example/demo applications, and this could _easily_ be a great demo.

**Comment by gavinking** [comment]
Alright, so as a first step I have ported almost all the Java code to Ceylon and cleaned it up somewhat. It's now using `ceylon.json` instead of the previous Java lib for json, among other changes.

But I need help to fix the build. I don't even understand who is driving the build, since there seem to be bits of both maven and gradle!

We need to wire in the ceylon compiler to the build, and then figure out something to repackage into the war, either using `ceylon war`, or perhaps simply unzipping the `.car` into the existing `classes` directory which seems to be where the compiled Java code goes now.

Who can help me out with this? @quintesse @chochos ?

**Comment by gavinking** [comment]
P.S. it's all on branch 83.

**Comment by quintesse** [comment]
Well like I said on Gitter, the official site is running on OpenShift which uses Maven to build everything automatically once you push the code to their Git repository. So it's not something we control directly. Perhaps the Ceylon cartridge is the way to go, I don't know, perhaps @matejonnet could give us a hand.

**Comment by gavinking** [comment]
Well look, as a first step we don't need it to run on OpenShift. If we could just get it running on a local wildfly, via `ceylon war`, that would be a *great* start, and surely pretty easy.

**Comment by quintesse** [comment]
Sure but remember that this means that as long as it doesn't work on OpenShift you can't push any of your changes to master, because what's in Git is literally what gets built.
So if you want to do all this in a separate branch or something, well be my guest :)

**Comment by gavinking** [comment]
Well sure I understand that. It's already on a branch.

**Comment by FroMage** [comment]
We do have a working Ceylon cartridge for openshift, even an openshift module and CLI plugin ;)

**Comment by gavinking** [comment]
OK, that sounds good, but since I don't know anything about OpenShift, I have _no clue_ how to turn the current project into a Ceylon openshift module.

**Comment by gavinking** [comment]
Wait, our OpenShift cartridge doesn't support WildFly, does it? It uses `ceylon.net` AFAIK.

So either:

1. I have to fix the current maven/gradle build to somehow call the Ceylon compiler, or
2. I have to stop using Servlets and change it to use `ceylon.net`.

Which is it going to be?

**Comment by FroMage** [comment]
The openshift Ceylon cartridge is documented, iirc you deploy sources, not compiled things. If you need wildfly just use the war CLI plugin and the wildfly cartridge.

**Comment by gavinking** [comment]
@FroMage it is set up to use the wildfly cartridge right now. But of course the wildfly cartridge doesn't know how to compile Ceylon code or invoke `ceylon war`.

**Comment by FroMage** [comment]
Oh I see, because there too you push sources. Just put some shell script in .openshift/bin or hooks or I don't know what. It's documented too.

**Comment by gavinking** [comment]
A _shell script_! That calls what? The `ceylon` command? And what, I also check a whole ceylon distribution into git?!

**Comment by FroMage** [comment]
Sure, why not?

**Comment by quintesse** [comment]
Btw, @gavinking @chochos , it was suggested that we could start translating the GitHub,js file to Ceylon, and I think you were thinking about doing it, weren't you @chochos ? 
But since I wrote it I wondered if you mind that I take it?
Thing is right now I'm busy with the OBR so I won't have time for it until somewhere next week or so.
So if you want to convert something to JS please start with something else :)

**Comment by chochos** [comment]
Go ahead @quintesse I'm porting `repl.js`
